lib/Cake/Test/Case/Model/Datasource/Database/MysqlTest.php

```diff
@@ -389,6 +389,36 @@ public function testBuildColumn() {
 		$expected = '`testName`  CHARACTER SET utf8 COLLATE utf8_unicode_ci DEFAULT NULL';
 		$this->assertEquals($expected, $result);
 		$this->Dbo->columns = $restore;
+
+		$data = array(
+			'name' => 'testName',
+			'type' => 'integer',
+			'storage' => 1
+		);
+		$result = $this->Dbo->buildColumn($data);
+		$expected = '`testName` tinyint(4)';
+		$this->assertEquals($expected, $result);
+		$this->Dbo->columns = $restore;
+
+		$data = array(
+			'name' => 'testName',
+			'type' => 'integer',
+			'storage' => 2
```

Comment:
How does this new storage option interact with `length`? Instead of tying the sizes to specific bytes, could we do something like:

```php
'my_int' => ['type' => 'integer', 'length' => 4, 'storageType' => 'small']
```

Another option could be `'storageSize' => 2` if byte are better. My concern with using specific bytes, is that we have no error handling for values like '3' or '6', and people will make mistakes that result in an incorrect schema being created. With string names, we can have a lookup of supported values and raise errors on unsupported values.